### DISCUSSION TITLE
How to delete/kill the Reference of a TWindow/TDialog

### DISCUSSION THREAD

**Discussion by paule32** [magiblot/tvision#145] [Q&A] [answered]
Hello,
I would like to delete/kill the Reference of a TWindow/TDialog.
When I use "delete this" in a TWindow/TDialog member, the Application tend to Crash, and/or freeze the Window.
When I use TObject::destroy(this); the Application tends to freeze, and Crash, too.

I am not sure, what happend after I call the close(); member of TWindow/TDialog.
The following Code, I use to initialize the TDialog Window:
TMyTvInfoWindow * info_message_window = nullptr;
// this member will be call when I open a file to display
void TFormApp::create_info_window()
{
    // here, I would like check, if the user close the window
    if (info_message_window != nullptr)
    return;

    info_message_window = (TMyTvInfoWindow*) validView(
    new TMyTvInfoWindow(
        TRect(    2,
        size.y - 10,
        size.x -  4,
        size.y -  3)));

    if (info_message_window != nullptr) {
        //info_message_window->helpCtx = hcCalendar;
        deskTop->insert( info_message_window );
    }
}

**Comment by magiblot** [comment]
When you delete an object, all references to it become invalid, meaning that you must make sure not use these references at all.

In this case you would either have to make sure that the `info_message_window` field gets set back to `nullptr` when closing the window, or simply avoid using such kind of mutable references, which is what the Turbo Vision manual suggests (page 168, Chapter 9: *Event-driven programming*):

![Screenshot_20240201_135448](https://github.com/magiblot/tvision/assets/20713561/fd22bab8-82d1-4adf-a758-0ccefe25d5cc)
![Screenshot_20240201_135502](https://github.com/magiblot/tvision/assets/20713561/9c215ce1-0fd9-4a8b-be41-497b370049f7)

**Comment by magiblot** [comment, marked as answer]
When you delete an object, all references to it become invalid, meaning that you must make sure not use these references at all.

In this case you would either have to make sure that the `info_message_window` field gets set back to `nullptr` when closing the window, or simply avoid using such kind of mutable references, which is what the Turbo Vision manual suggests (page 168, Chapter 9: *Event-driven programming*):

![Screenshot_20240201_135448](https://github.com/magiblot/tvision/assets/20713561/fd22bab8-82d1-4adf-a758-0ccefe25d5cc)
![Screenshot_20240201_135502](https://github.com/magiblot/tvision/assets/20713561/9c215ce1-0fd9-4a8b-be41-497b370049f7)
